Allow passing UV coordinates as Vec2f

yyny opened the pull request:
Rotating UV coordinates using vector math becomes extremely annoying when every `sprite` call requires the individual `u` and `v` coordinates to be extracted again.

This change will allow passing UV coordinates in a container that will always be available.

yyny committed:
Allow passing UV coordinates as Vec2f

Rotating UV coordinates using vector math becomes extremely annoying when every `sprite` call requires the individual `u` and `v` coordinates to be extracted again.

This change will allow passing UV coordinates in a container that will always be available.

Player3324 reviewed (approved):
Why not I guess, I hope nobody will create Vec2fs specifically to call this though.

yyny commented:
Great concern, we can add a javadoc comment if this is an issue, but I doubt many people will stumble upon this method unless they already have a vector they want to pass.

i509VCB reviewed (approved):
Please do add the javadoc to that method about allocation.

yyny committed:
Revert b0fe27e0d78678fb2aece2271c80639287fe62b5

yyny committed:
Allow Vec2f UVs in MutableQuadView

This seems like the better place to put this.

yyny committed:
Fix return type

oops

yyny committed:
Add override in QuadEmitter

yyny committed:
Fix JavaDoc styling

yyny committed:
Duplicate JavaDoc and function definition

There is probably a way in Java to override the method while keeping the JavaDoc, but I can't be bothered, and this way, the user gets a more relevant JavaDoc anyway.

yyny committed:
Remove Trailing Whitespace

I'm pretty sure QuadEmitter allowed trailing whitespace, though...

yyny committed:
Fix JavaDoc styling in QuadEmitter

Also removes trailing whitespace

grondag commented:
As Player already noted,  I wouldn't allocate a `Vec2f` just for this but if you already have one, it's handy enough.

I suggest removing the `spriteIndex` parameter (use 0) because Indigo does not support it, FRAPI defines no behavior for it, and I've removed support for it in FREX/Canvas as well.

It was convenient to use when it was supported, but the lack of consistency was creating unnecessary complications.  I intend to submit a PR to deprecate and replace the methods that accept `spriteIndex` and then hopefully remove them in the 1.17 cycle.
